### PR TITLE
Non-text colour contrast for WCAG 2.1

### PR DESCRIPTION
**Pull request by AdamChrimes**
### Colour palette updates

| Colour name | Old value | Replace with|
| ------------- | ------------- | ------------- |
| $color_nhsuk-grey-1  | #425563 | #4c6272 |
| $color_nhsuk-red  | #da291c | #d5281b |

### Colour variables removed (💥 breaking change)

| Colour variable removed  | Suggested replacement |
| ------------- | ------------- |
| $color_tint_nhsuk-warm-yellow-30  | $color_nhsuk-warm-yellow  |
| $color_tint_nhsuk-warm-yellow-10  | $color_nhsuk-warm-yellow  |
| $nhsuk-link-focus-color | $nhsuk-focus-text-color |
| $nhsuk-link-hover-background-color | $nhsuk-focus-color |
| $nhsuk-link-focus-background-color | $nhsuk-focus-color |
| $nhsuk-link-active-background-color | $nhsuk-focus-color |
| $nhsuk-button-text-colour | $nhsuk-button-text-color |
| $nhsuk-button-hover-colour | $nhsuk-button-hover-color |
| $nhsuk-reverse-button-text-colour | $nhsuk-reverse-button-text-color |
| $nhsuk-button-shadow-colour | $nhsuk-button-shadow-color |
| $nhsuk-secondary-button-colour | $nhsuk-secondary-button-color |
| $nhsuk-secondary-button-hover-colour | $nhsuk-secondary-button-hover-color |
| $nhsuk-secondary-button-shadow-colour | $nhsuk-secondary-button-shadow-color |
| $nhsuk-reverse-button-colour | $nhsuk-reverse-button-color |
| $nhsuk-reverse-button-hover-colour | $nhsuk-reverse-button-hover-color |
| $nhsuk-reverse-button-shadow-colour | $nhsuk-reverse-button-shadow-color |

### Colour variables added (:new: new feature)

`$nhsuk-button-active-color: shade($nhsuk-button-color, 50%);`
`$nhsuk-secondary-button-active-color: shade($nhsuk-secondary-button-color, 50%);`
`$nhsuk-reverse-button-active-color: shade($nhsuk-reverse-button-color, 50%);`

#### Spelling corrections

- `$nhsuk-button-color`
- `$nhsuk-button-hover-color`
- `$nhsuk-secondary-button-color`
- `$nhsuk-secondary-button-hover-color`
- `$nhsuk-secondary-button-shadow-color`
- `$nhsuk-reverse-button-color`
- `$nhsuk-reverse-button-hover-color`
- `$nhsuk-reverse-button-shadow-color`
- `$nhsuk-focus-color`
- `$nhsuk-focus-text-color`
- `$nhsuk-button-shadow-color`

### Mixins removed (💥 breaking change)

| Colour variable removed  | Suggested replacement |
| ------------- | ------------- |
| `@mixin nhsuk-link-style-focus` | `@mixin nhsuk-focused-text` |
| `@mixin nhsuk-focusable` | `@mixin nhsuk-focused-text` |
| ` @mixin nhsuk-focusable-fill` | `@mixin nhsuk-focused-text` |

### Mixins added (:new: new feature)

- `@mixin nhsuk-focused-text`
- `@mixin nhsuk-focused-input`
- `@mixin nhsuk-focused-button`

## Related issue

https://github.com/nhsuk/nhsuk-frontend/issues/473

## Checklist

- [ ] Tested against our [testing policy](https://github.com/nhsuk/nhsuk-frontend/blob/master/docs/contributing/testing.md) (Resolution, Browser & Accessibility)
- [ ] Follows our [coding standards and style guide](https://github.com/nhsuk/nhsuk-frontend/blob/master/docs/contributing/coding-standards.md)
- [ ] CHANGELOG entry

### CODE REVIEW

File: app/_templates/layout.njk [hunk 1]
```diff
@@ -9,7 +9,7 @@
     <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
     <meta name="description" content="">
 
-    <title>NHS.UK frontend - {{ title }}</title>
+    <title>{{ title }} | NHS.UK frontend</title>
```
Comment (AdamChrimes) [inline app/_templates/layout.njk]: This should probably be 'Page title - NHS.UK frontend' to match the guidance in the service manual ([https://beta.nhs.uk/service-manual/accessibility/development#set-page-titles](https://beta.nhs.uk/service-manual/accessibility/development#set-page-titles))

File: app/_templates/page.njk [hunk 2]
```diff
@@ -12,7 +47,7 @@
   {{ header({
     "showNav": "false",
     "showSearch": "false",
-    "homeHref": "https://www.nhs.uk"
+    "homeHref": "/"
```
Comment (AdamChrimes) [inline app/_templates/page.njk]: This won't work due to the GitHub pages being hosted at `/nhsuk-frontend/` using `/` would return you to `https://nhsuk.github.io/` - which gives an error page. We had this problem in the past, which is why we used the nhs.uk website URL. Using `/nhsuk-frontend` would fix it on the GitHub pages but this wouldn't work locally as we work at the root directory.

File: app/_templates/page.njk [hunk 3]
```diff
@@ -52,9 +87,25 @@
 
   {{ footer({
     "links": [
+      {
+        "URL": "/",
```
Comment (AdamChrimes) [inline app/_templates/page.njk]: See above comment

File: app/pages/examples.njk [hunk 4]
```diff
@@ -10,16 +10,25 @@
...
+    text: "NHS.UK frontend"
   }) }}
 {% endblock %}
 
 {% block content %}
 
   <h1>Examples</h1>
 
-  <h2>Components</h2>
+  <h2>Test pages</h2>
```
Comment (AdamChrimes) [inline app/pages/examples.njk]: Should these be before the components list? Can we utilize the right-hand side of the page and have half-half?

File: app/test-pages/amd/getting-diagnosed.njk [hunk 5]
```diff
@@ -0,0 +1,144 @@
+{% set title = 'Getting diagnosed | Age-related macular degeneration (AMD)  | Test page' %}
```
Comment (AdamChrimes) [inline app/test-pages/amd/getting-diagnosed.njk]: See comment about page titles, should be 'Page title - Website', we don't include page sections.

File: app/test-pages/amd/index.njk [hunk 6]
```diff
@@ -0,0 +1,83 @@
+{% set title = 'Overview | Age-related macular degeneration (AMD)  | Test page' %}
```
Comment (AdamChrimes) [inline app/test-pages/amd/index.njk]: See comment about titles

File: app/test-pages/amd/living-with-amd.njk [hunk 7]
```diff
@@ -0,0 +1,144 @@
+{% set title = 'Living with AMD | Age-related macular degeneration (AMD)  | Test page' %}
```
Comment (AdamChrimes) [inline app/test-pages/amd/living-with-amd.njk]: See comment about page titles

File: app/test-pages/amd/symptoms.njk [hunk 8]
```diff
@@ -0,0 +1,136 @@
+{% set title = 'Symptoms | Age-related macular degeneration (AMD)  | Test page' %}
```
Comment (AdamChrimes) [inline app/test-pages/amd/symptoms.njk]: See comment about page titles

File: app/test-pages/amd/treatment.njk [hunk 9]
```diff
@@ -0,0 +1,154 @@
+{% set title = 'Treatments | Age-related macular degeneration (AMD)  | Test page' %}
```
Comment (AdamChrimes) [inline app/test-pages/amd/treatment.njk]: See comment about page titles

File: app/test-pages/ankle-pain.njk [hunk 10]
```diff
@@ -0,0 +1,175 @@
+{% set title = 'Ankle pain | Test page' %}
```
Comment (AdamChrimes) [inline app/test-pages/ankle-pain.njk]: See comment about page titles

File: app/test-pages/chickenpox.njk [hunk 11]
```diff
@@ -0,0 +1,182 @@
+{% set title = 'Chickenpox | Test page' %}
```
Comment (AdamChrimes) [inline app/test-pages/chickenpox.njk]: See comment about page titles

File: app/test-pages/health-a-to-z.njk [hunk 12]
```diff
@@ -0,0 +1,656 @@
+{% set title = 'Health A to Z | Test page' %}
```
Comment (AdamChrimes) [inline app/test-pages/health-a-to-z.njk]: See comment about page titles

File: app/test-pages/newsletter.njk [hunk 13]
```diff
@@ -0,0 +1,108 @@
+{% set title = 'Newsletter signup | Test page' %}
```
Comment (AdamChrimes) [inline app/test-pages/newsletter.njk]: See comment about page titles

File: app/test-pages/nhsuk-homepage.njk [hunk 14]
```diff
@@ -0,0 +1,262 @@
+{% set title = 'The NHS website home page | Test page' %}
```
Comment (AdamChrimes) [inline app/test-pages/nhsuk-homepage.njk]: See comment about page titles

File: packages/core/settings/_colours.scss [hunk 15]
```diff
@@ -20,7 +20,7 @@ $color_nhsuk-blue: #005eb8;
 $color_nhsuk-white: #ffffff;
 $color_nhsuk-black: #212b32;
 $color_nhsuk-green: #007f3b;
-$color_nhsuk-red: #da291c;
+$color_nhsuk-red: #d5281b;
```
Comment (AdamChrimes) [inline packages/core/settings/_colours.scss]: We need to document this change and update the service manual

File: packages/core/settings/_colours.scss [hunk 16]
```diff
@@ -36,7 +36,7 @@ $color_nhsuk-aqua-green: #00A499;
 // Greyscale
 //
 
-$color_nhsuk-grey-1: #425563;
+$color_nhsuk-grey-1: #566f82;
```
Comment (AdamChrimes) [inline packages/core/settings/_colours.scss]: We need to document this change and update the service manual

File: packages/core/settings/_colours.scss [hunk 17]
```diff
@@ -92,39 +90,34 @@ $color_shade_nhsuk-blue-50: shade($color_nhsuk-blue, 50%); // [4] //
 $color_shade_nhsuk-green-35: shade($color_nhsuk-green, 35%); // [5] //
 $color_shade_nhsuk-green-50: shade($color_nhsuk-green, 50%); // [6] //
 
-$color_tint_nhsuk-warm-yellow-30: tint($color_nhsuk-warm-yellow, 30%); // [7] //
```
Comment (AdamChrimes) [inline packages/core/settings/_colours.scss]: We need to document this change and update the service manual.

Also need document which colour variable people should use instead.

File: packages/core/settings/_colours.scss [hunk 18]
```diff
@@ -92,39 +90,34 @@ $color_shade_nhsuk-blue-50: shade($color_nhsuk-blue, 50%); // [4] //
...
-$color_tint_nhsuk-warm-yellow-10: tint($color_nhsuk-warm-yellow, 10%); // [8] //
```
Comment (AdamChrimes) [inline packages/core/settings/_colours.scss]: We need to document this change and update the service manual.

Also need document which colour variable people should use instead.

File: packages/core/tools/_focused.scss [hunk 19]
```diff
@@ -0,0 +1,54 @@
...
+// ==========================================================================
+
+//
+// Focused text
+//
+// Provides an outline to clearly indicate when the target element is focused.
+// Used for interactive text-based elements.
+//
+
+@mixin nhsuk-focused-text {
```
Comment (AdamChrimes) [inline packages/core/tools/_focused.scss]: We need to document this new mixin somewhere

File: packages/core/tools/_links.scss [hunk 20]
```diff
@@ -109,25 +115,6 @@
...
   }
 }
-
-//
-// Default link focus only styling
-//
-// Usage: @include nhsuk-link-style-focus;
-//
-
-@mixin nhsuk-link-style-focus {
```
Comment (AdamChrimes) [inline packages/core/tools/_links.scss]: Document the removal of this mixin and suggest an alternative (if there is one)

File: packages/components/details/_details.scss [hunk 21]
```diff
@@ -137,24 +137,38 @@
...
     position: relative;
   }
 
   .nhsuk-details__text {
     @include nhsuk-responsive-padding(4);
+
     border-left: 0;
     margin-left: 0;
     margin-top: 0;
+    padding-top: 0 !important;
```
Comment (AdamChrimes) [inline packages/components/details/_details.scss]: Use of `!important` here, can this be avoided? If not can we add an eslint disable to this line and a comment as to why.

File: packages/components/header/_autocomplete.scss [hunk 22]
```diff
@@ -50,7 +50,11 @@
     width: 100%; /* [3] */
 
     &:focus {
-      box-shadow: inset 0 0 0 $nhsuk-box-shadow-spread $nhsuk-focus-color;
+      border: none;
```
Comment (AdamChrimes) [inline packages/components/header/_autocomplete.scss]: Duplicate border styles here (duplicated on line 54), which one is the correct one?

If its `border: none`, this needs to be `border: 0` instead.

**Comment by AdamChrimes** [comment]
Is there a risk in publishing the test pages with real conditions content to a public domain without a password? People may think they are the actual pages if they stumble across them (especially with the NHS logo linking to a homepage)

**Comment by AdamChrimes** [comment]
Version number needs bumping to `3.0.0` and a `CHANGELOG` entry adding

File: app/_templates/page.njk (hunk 2 above)
Comment (davidhunter08) [reply app/_templates/page.njk]: Replaced `/` with `/nhsuk-frontend/`

File: app/_templates/page.njk (hunk 3 above)
Comment (davidhunter08) [reply app/_templates/page.njk]: Replaced `/` with `/nhsuk-frontend/`

File: packages/components/details/_details.scss (hunk 21 above)
Comment (davidhunter08) [reply packages/components/details/_details.scss]: Refactored `padding-top: 0 !important;`

File: packages/components/header/_autocomplete.scss (hunk 22 above)
Comment (davidhunter08) [reply packages/components/header/_autocomplete.scss]: Removed `border: none;`

**Comment by AdamChrimes** [comment]
Need to look at the hover and focus state in the immediate care card

**Comment by davidhunter08** [comment]
- [ ] Check the skip link component

**Comment by AdamChrimes** [comment]
- [ ] Need to review the broken BackstopJS tests, most of the broken tests are focus styles however there are some others which don't relate to focus styles.
- [ ] Need to add a CHANGELOG entry under 3.0.0

**Pull request merged**